### PR TITLE
SPARK-53 fix silly @require_channel decorator requirement

### PR DESCRIPTION
**Pull request by theunkn0wn1**
This was a bit of a doozy.

As described in SPARK-53 there was a bug that required the `@require_channel` and `@require_dm` to be used as follows:
```py
@require_channel()
async def my_command(...)
```

the empty `()` being an anoying requirement. This PR fixes that.

the valid usages for the decorators has been corrected, can can be used in either of the following forms
```py
@require_channel
async def my_command(...)
```
which will use the default access denied message.

```py
@require_channel(message="Heck no.")
async def my_command(...)
```
The kwarg **must be specified**, if the kwarg is not specified it will be ignored and the function may be have unexpectidly. 

Fixes SPARK-53

### CODE REVIEW

File: Modules/permissions.py [hunk 1]
```diff
@@ -131,7 +132,7 @@ def real_decorator(func):
     return real_decorator
 
 
-def require_channel(message: str = "This command must be invoked in a channel."):
+def require_channel(*args, **kwargs):
```
Comment (MHajoha) [inline Modules/permissions.py]: It should still be possible to have a nice signature without varargs here. Simply have one parameter which can be either the message or the function. The same goes for `require_dm`.

Comment (theunkn0wn1) [reply Modules/permissions.py]: Oh, and `len(args)` can also be zero, which is a key reason i cannot have a clean signature as python wouldn't accept it.
```py
@require_channel
async def...
```
would invoke `require_channel` with zero arguments

```py
@require_channel()
```
would invoke `require_channel` with exactly one argument: the function
```py
@require_channel(message="potato")
```
would invoke require_channel` with exactly one argument and exactly one kwarg

Comment (MHajoha) [reply Modules/permissions.py]: `require_channel(message: Union[str, Callable] = "default message")` would work fine. To go through the options:
* `@require_channel` would pass the function as the first and only argument (i.e. message).
* `@require_channel()` would use the default message.
* `@require_channel(message="message")` and `@require_channel("message")` both pass the provided message.

You would then merely need to test whether `message` is a function or a string.

Comment (MHajoha) [reply Modules/permissions.py]: For an example, see `pytest.fixture`.

**Comment by theunkn0wn1** [comment]
Signature fixed.

**Pull request merged**